pkg/disttask/framework/storage/task_table.go

```diff
@@ -784,7 +823,142 @@ func (stm *TaskManager) ResumeSubtasks(ctx context.Context, taskID int64) error
 	return err
 }
 
+// SwitchTaskStep implements the dispatcher.TaskManager interface.
+func (stm *TaskManager) SwitchTaskStep(
+	ctx context.Context,
+	task *proto.Task,
+	nextState proto.TaskState,
+	nextStep proto.Step,
+	subtasks []*proto.Subtask,
+) error {
+	return stm.WithNewTxn(ctx, func(se sessionctx.Context) error {
+		vars := se.GetSessionVars()
+		if vars.MemQuotaQuery < variable.DefTiDBMemQuotaQuery {
+			bak := vars.MemQuotaQuery
+			if err := vars.SetSystemVar(variable.TiDBMemQuotaQuery,
+				strconv.Itoa(variable.DefTiDBMemQuotaQuery)); err != nil {
+				return err
+			}
+			defer func() {
+				_ = vars.SetSystemVar(variable.TiDBMemQuotaQuery, strconv.Itoa(int(bak)))
+			}()
+		}
+		err := stm.updateTaskStateStep(ctx, se, task, nextState, nextStep)
+		if err != nil {
+			return err
+		}
+		if vars.StmtCtx.AffectedRows() == 0 {
```

Comment:
what if the insert operation failed at the new owner side?

Reply:
if other owner update task step success, we can return directly here